web_src/js/markup/tasklist.js

```diff
@@ -55,12 +55,11 @@ export function initMarkupTasklist() {
           const updateUrl = editContentZone.getAttribute('data-update-url');
           const context = editContentZone.getAttribute('data-context');
 
-          await $.post(updateUrl, {
-            ignore_attachments: true,
-            _csrf: window.config.csrfToken,
-            content: newContent,
-            context
-          });
+          const requestBody = new FormData();
+          requestBody.append('ignore_attachments', 'true');
+          requestBody.append('content', newContent);
+          requestBody.append('context', context);
+          await POST(updateUrl, {body: requestBody});
```

Comment:
```suggestion
          await POST(updateUrl, {data: requestBody});
```

Reply:
As per https://github.com/go-gitea/gitea/pull/29170#discussion_r1490730826. @wxiaoguang agree?

Reply:
Please decide and push it then merge

Reply:
Will wait on @wxiaoguang, merging now would be rude 😆

Reply:
Never mind, either looks good to me. Since the framework was designed by you, so just do as your plan.

The only question in my mind is still that why there is a "body", when it should be used, and why "data" fallbacks to "body" in the implementation.

Reply:
I agree, we can remove the body argument from the arguments, it's just confusing that there are two ways to pass the data.